managed/services/telemetry/telemetry.go

```diff
@@ -208,17 +208,22 @@ func (s *Service) processSendCh(ctx context.Context) {
 }
 
 func (s *Service) prepareReport(ctx context.Context) *pmmv1.ServerMetric {
+	dataSources := s.locateDataSources(s.config.telemetry)
 	telemetryMetric, _ := s.makeMetric(ctx)
-
 	var totalTime time.Duration
-	for _, telemetry := range s.config.telemetry {
-		// locate DS
-		ds, err := s.LocateTelemetryDataSource(telemetry.Source)
+
+	for _, dataSource := range dataSources {
+		err := dataSource.PreFetch(ctx)
 		if err != nil {
```

Comment:
if `init` for a DS is failing should be consider that this DS should not be used for metrics fetching?

Reply:
Fixed.